client/retryable_client.go

```diff
@@ -0,0 +1,279 @@
+		}
+		res = r
+		return nil
+	})
+	return res, err
+}
+
+// GetProviderChannel returns the provider channel
+func (bwr *BlockchainWithRetries) GetProviderChannel(accountantAddress, addressToCheck common.Address) (ProviderChannel, error) {
+	var res ProviderChannel
+	err := bwr.callWithRetry(func() error {
+		r, err := bwr.bc.GetProviderChannel(accountantAddress, addressToCheck)
+		if err != nil {
+			return errors.Wrap(err, "could not get provider channel")
+		}
+		res = r
+		return nil
+	})
+
+	return res, err
+}
+
+// SubscribeToPromiseSettledEvent subscribes to promise settled events
+func (bwr *BlockchainWithRetries) SubscribeToPromiseSettledEvent(providerID, accountantID common.Address) (chan *bindings.AccountantImplementationPromiseSettled, func(), error) {
+	var sink chan *bindings.AccountantImplementationPromiseSettled
+	var cancel func()
+	err := bwr.callWithRetry(func() error {
+		s, c, err := bwr.bc.SubscribeToPromiseSettledEvent(providerID, accountantID)
+		if err != nil {
+			return errors.Wrap(err, "could not subscribe to settlement events")
+		}
+		sink = s
+		cancel = c
+		return nil
+	})
+	return sink, cancel, err
+}
+
+// SubscribeToConsumerBalanceEvent subscribes to the consumer balance change events
+func (bwr *BlockchainWithRetries) SubscribeToConsumerBalanceEvent(channel, mystSCAddress common.Address) (chan *bindings.MystTokenTransfer, func(), error) {
+	var sink chan *bindings.MystTokenTransfer
+	var cancel func()
+	err := bwr.callWithRetry(func() error {
+		s, c, err := bwr.bc.SubscribeToConsumerBalanceEvent(channel, mystSCAddress)
+		if err != nil {
+			return errors.Wrap(err, "could not subscribe to settlement events")
+		}
+		sink = s
+		cancel = c
+		return nil
+	})
+	return sink, cancel, err
+}
+
+// IsRegistered checks wether the given identity is registered or not
+func (bwr *BlockchainWithRetries) IsRegistered(registryAddress, addressToCheck common.Address) (bool, error) {
+	var res bool
+	err := bwr.callWithRetry(func() error {
+		r, err := bwr.bc.IsRegistered(registryAddress, addressToCheck)
+		if err != nil {
+			return errors.Wrap(err, "check registration status")
+		}
+		res = r
+		return nil
+	})
+	return res, err
+}
+
+// GetMystBalance returns the balance in myst
+func (bwr *BlockchainWithRetries) GetMystBalance(mystSCAddress, channel common.Address) (*big.Int, error) {
+	var res *big.Int
+	err := bwr.callWithRetry(func() error {
+		result, bcErr := bwr.bc.GetMystBalance(mystSCAddress, channel)
+		if bcErr != nil {
+			return errors.Wrap(bcErr, "could not get myst balance")
+		}
+		res = result
+		return nil
+	})
+	return res, err
+}
+
+// GetRegistrationFee returns the registration fee
+func (bwr *BlockchainWithRetries) GetRegistrationFee(registryAddress common.Address) (*big.Int, error) {
+	var res *big.Int
+	err := bwr.callWithRetry(func() error {
+		result, bcErr := bwr.bc.GetRegistrationFee(registryAddress)
+		if bcErr != nil {
+			return errors.Wrap(bcErr, "could not get registration fee")
```

Comment:
In case of error it will be duplicated.

```
could not get registration fee: could not get registration fee: ```
```

As you already wrapped errors inside Blockchain maybe it could be simpler.

```go
func (bwr *BlockchainWithRetries) GetRegistrationFee(registryAddress common.Address) (*big.Int, error) {
	var res *big.Int
	err := bwr.callWithRetry(func() error {
                var bcErr error
		res, bcErr = bwr.bc. GetRegistrationFee(registryAddress)
		return bcErr
	})
	return res, err
}

```